pkg/api/common/versions.go

```diff
@@ -60,19 +65,110 @@ func GetAllSupportedKubernetesVersions() []string {
 }
 
 // GetVersionsGt returns a list of versions greater than a semver string given a list of versions
-func GetVersionsGt(versions []string, version string) []string {
+// inclusive=true means that we test for equality as well
+// preReleases=true means that we include pre-release versions in the list
+func GetVersionsGt(versions []string, version string, inclusive, preReleases bool) []string {
 	// Try to get latest version matching the release
 	var ret []string
+	var cons *semver.Constraints
+	minVersion, _ := semver.NewVersion(version)
 	for _, v := range versions {
 		sv, _ := semver.NewVersion(v)
-		cons, _ := semver.NewConstraint(">" + version)
+		if preReleases && minVersion.Prerelease() == "" {
+			sv, _ = semver.NewVersion(fmt.Sprintf("%d.%d.%d", sv.Major(), sv.Minor(), sv.Patch()))
+		}
+		if inclusive {
+			cons, _ = semver.NewConstraint(">=" + version)
+		} else {
+			cons, _ = semver.NewConstraint(">" + version)
+		}
 		if cons.Check(sv) {
 			ret = append(ret, v)
 		}
 	}
 	return ret
 }
 
+// GetVersionsLt returns a list of versions greater than a semver string given a list of versions
```

Comment:
Is it just me or should this function be called GetVersionsGt?

Reply:
scratch that, the comment should be returns a list of versions *lesser*  than a semver string